plugins/hw_event/hw_event_plugin.go

```diff
@@ -200,10 +114,48 @@ func publishHwEvent(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	for i, e := range redfishEvent.Events {
+		if e.Message == "" {
+			parsed, err := parseMessage(e)
+			if err == nil {
+				redfishEvent.Events[i] = parsed
+			} else {
+				// ignore error
+				log.Debugf("error parsing message: %v", err)
+			}
+
+		}
+	}
+
 	data := v1hwevent.CloudNativeData()
 	data.SetVersion("v1") //nolint:errcheck
 	data.SetData(&redfishEvent)
 
-	event, _ := common.CreateHwEvent(pub.ID, "HW_EVENT", data)
-	_ = common.PublishHwEvent(scConfig, event)
+	event, _ := common.CreateHwEvent(pub.ID, "HW_EVENT", data, timeReceived)
+	_ = common.PublishHwEventViaAPI(scConfig, event)
+}
+
+func parseMessage(m hwevent.EventRecord) (hwevent.EventRecord, error) {
+	addr := "localhost:9999"
```

Comment:
can you explain why port is hardcoded? also move it to a constant

Reply:
This is temporary. I have a followup story to put message-parser in container, and will pass port number from environment variable.

Reply:
move the string to a constant

Reply:
done